### PR TITLE
Fixed #6775: Allow non-standard letters to be use in equations

### LINKED ISSUE #6775: Allow non-standard letters to be use in equations
This is just being created in order to track the discussion in #6773 about allowing non-standard letters to be used in equations.

@rsudev commented:
>A-Z (if that really is the definition) is less ambiguous than LETTER (which in some places might end up being äöü or even åđžć ...

and @UniQP said:
>I would prefer to allow all characters accepted by Character.isLetterOrDigit(...) and simply use letter and digit in the message. Is that feasible?

And later on @pstorch commented:
>Regarding the letter and digit discussion: at the moment it is ambiguous. Just tried ö, which is a letter, and got the above error message. I personally think we should restrict the variable letters to A-Z. If we allow all unicode letters from all languages, then we might run into sorting issues again. So, to avoid that and make things simple, I would suggest A-Z letters only. What do you think?

### PR DESCRIPTION
**Pull request by S-Bartfast**
Fixed #6775: Allow non-standard letters to be use in equations

### CODE REVIEW

**Comment by Lineflyer** [comment]
retest this please

**Comment by pstorch** [comment]
@S-Bartfast there is a conflict now. And please a more descriptive commit message.

**Comment by S-Bartfast** [comment]
Hmm, I'm not really sure where the conflicts have come from but I've resolved the conflicts and rebased onto master but the change doesn't seem to be coming through here.  I may have to issue another pull request.

**Comment by S-Bartfast** [comment]
Okay, I think it's good now.
That was a bit confusing??

**Comment by S-Bartfast** [comment]
Okay, think I've got it!

File: main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java [hunk 1]
```diff
@@ -580,17 +586,34 @@ private void setButtonInputValuesFromGP() {
...
-
         final String lat = getLatResult();
         final String lon = getLonResult();
 
         if (lat.contains("_") || lat.contains("*") || lon.contains("_") || lon.contains("*")) {
             return false;
         }
 
-        try {
-            current = new Geopoint(lat, lon);
```
Comment (pstorch) [inline main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java]: @S-Bartfast wasn't this just changed the other way around?

File: main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java [hunk 2]
```diff
@@ -843,7 +866,7 @@ private String getValues(final List<CalculateButton> buttons, final int degDigit
         for (final EditButton button : buttons) {
             // Remove inactive and blank digits from result
             if (button.getVisibility() == View.VISIBLE) {
-                if (button.getLabel() == CalculateButton.ButtonData.BLANK) {
```
Comment (pstorch) [inline main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java]: Same here

File: main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java [hunk 3]
```diff
@@ -1100,22 +1123,24 @@ private CalculatorVariable getVariable(final char name, final List<CalculatorVar
...
-                                                   final Range range,
+                                                   final CaseCheck theCase,
                                                    final String hintText,
                                                    final TextWatcher textWatcher) {
         final List<CalculatorVariable> returnList = new ArrayList<>();
 
         final char[] sortedVariables = variableNames.toCharArray();
         Arrays.sort(sortedVariables);
 
-        for (final char ch : sortedVariables) {
```
Comment (pstorch) [inline main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java]: also here

File: main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java (hunk 1 above)
Comment (S-Bartfast) [reply main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java]: Yeah, I've definitely managed to get something "arse about face" as we say in Australia.

Comment (S-Bartfast) [reply main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java]: Reverted.

File: main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java (hunk 2 above)
Comment (S-Bartfast) [reply main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java]: Reverted.

File: main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java (hunk 3 above)
Comment (S-Bartfast) [reply main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java]: Reverted.

Comment (pstorch) [reply main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java]: Hmm, I still see the wrong version of the loop :thinking:

Comment (S-Bartfast) [reply main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java]: Okay, that should get it.

**Comment by pstorch** [comment]
@S-Bartfast thanks :+1:

**Pull request merged**